notebooks/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-publish-and-run-using-rest-endpoint.ipynb

```diff
@@ -0,0 +1,469 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.compute_target import ComputeTargetException\n",
+    "\n",
+    "aml_compute_target = \"cpu-cluster\"\n",
+    "try:\n",
+    "    aml_compute = AmlCompute(ws, aml_compute_target)\n",
+    "    print(\"found existing compute target.\")\n",
+    "except ComputeTargetException:\n",
+    "    print(\"creating new compute target\")\n",
+    "    \n",
+    "    provisioning_config = AmlCompute.provisioning_configuration(vm_size = \"STANDARD_D2_V2\",\n",
+    "                                                                min_nodes = 1, \n",
+    "                                                                max_nodes = 4)    \n",
+    "    aml_compute = ComputeTarget.create(ws, aml_compute_target, provisioning_config)\n",
+    "    aml_compute.wait_for_completion(show_output=True, min_node_count=None, timeout_in_minutes=20)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# For a more detailed view of current Azure Machine Learning Compute status, use get_status()\n",
+    "# example: un-comment the following line.\n",
+    "# print(aml_compute.get_status().serialize())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Building Pipeline Steps with Inputs and Outputs\n",
+    "A step in the pipeline can take [dataset](https://docs.microsoft.com/python/api/azureml-core/azureml.data.filedataset?view=azure-ml-py) as input. This dataset can be a data source that lives in one of the accessible data locations, or intermediate data produced by a previous step in the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Uploading data to the datastore\n",
+    "data_path = def_blob_store.upload_files([\"./20news.pkl\"], target_path=\"20newsgroups\", overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reference the data uploaded to blob storage using file dataset\n",
+    "# Assign the datasource to blob_input_data variable\n",
+    "blob_input_data = Dataset.File.from_files(data_path).as_named_input(\"test_data\")\n",
+    "print(\"Dataset created\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define intermediate data using OutputFileDatasetConfig\n",
+    "processed_data1 = OutputFileDatasetConfig(name=\"processed_data1\")\n",
+    "print(\"Output dataset object created\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define a Step that consumes a dataset and produces intermediate data.\n",
+    "In this step, we define a step that consumes a dataset and produces intermediate data.\n",
+    "\n",
```

Comment:
this info should be in documentation, not examples

Reply:
Reduced info within notebook